### PR TITLE
removes unnecessarily adding the source to include_directories:

### PR DESCRIPTION
**Pull request by codingdave**
DefineCMakeDefaults contains set(CMAKE_INCLUDE_CURRENT_DIR ON) which
automatically lets CMake add its source and build directory to the include
directories.

https://cmake.org/cmake/help/v3.0/variable/CMAKE_INCLUDE_CURRENT_DIR.html
Automatically add the current source- and build directories to the include path.

If this variable is enabled, CMake automatically adds in each directory ${CMAKE_CURRENT_SOURCE_DIR} and ${CMAKE_CURRENT_BINARY_DIR} to the include path for this directory. These additional include directories do not propagate down to subdirectories. This is useful mainly for out-of-source builds, where files generated into the build tree are included by files located in the source tree.

By default CMAKE_INCLUDE_CURRENT_DIR is OFF.

-> Increased maintainability
-> Increased readability
-> Automatic consistent behavior by CMAKE_INCLUDE_CURRENT_DIR
-> one global place to tweak the compile behavior

~This is one change that I would like to have been be merged before I will
create a PR for darktable out of source builds~

### CODE REVIEW

**Comment by parafin** [comment]
Are you saying that out-of-source builds are not working? What do you mean by "out-of-source" then?

**Comment by codingdave** [comment]
@parafin I mean running the compiled darktable from the build directory is not working at the moment, also because the pathes are absolute. E.g. running from the build directory ends up with dt having `darktable.datadir = /usr/local/share/darktable`. This also holds for the icons and stuff. I have fixed that and hope my PR will be ok (upcoming).

**Comment by parafin** [comment]
Ah, that makes more sense:)

**Comment by codingdave** [comment]
Apart from the non-building state, that I apparently have to fix please **do not yet merge this PR**, I have just discovered, that I missed some other locations. I will fix that soon.

File: src/imageio/format/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
-include_directories("${CMAKE_CURRENT_BINARY_DIR}/../../" "${CMAKE_CURRENT_SOURCE_DIR}")
+include_directories("${CMAKE_CURRENT_BINARY_DIR}/../../")
```
Comment (cryptomilk) [inline src/imageio/format/CMakeLists.txt:1]: Maybe use `include_directories(${darktable_BINARY_DIR})`?

**Comment by cryptomilk** [comment]
```
/build/darktable/src/iop/filmicrgb.c:19:10: fatal error: 'config.h' file not found

#include "config.h"

         ^~~~~~~~~~

1 error generated.
```

I guess in `src/iop/CMakeLists.txt` you need `include_directories(${darktable_BINARY_DIR}/src)`

**Comment by TurboGit** [comment]
I'll let @parafin give the go on this one, I do not qualify to review complex cmake changes.

**Comment by parafin** [comment]
I'm fine with these changes, but of course they shouldn't break build, like they currently do. So fix CI and ping us here.

**Comment by cytrinox** [comment]
How should darktable find the data directory when running from build dir?

````
 cyt  @pollux  ~/.../darktable-master/build   (master)  ✚1  ✔0  ✘0  $  ./src/darktable --library dummy/foobar.db --cachedir dummy/cache --configdir dummy/config
[dt_codepaths_init] will be using HIGHLY EXPERIMENTAL plain OpenMP SIMD codepath.
[dt_codepaths_init] SSE2-optimized codepath is disabled or unavailable.
[dt_codepaths_init] expect a LOT of functionality to be broken. you have been warned.
[init] moving database into new XDG directory structure
dt_gui_load_theme: error parsing /usr/local/share/darktable/themes/darktable.css: <broken file>:1:0Failed to import: Error opening file /usr/local/share/darktable/themes/darktable.css: No such file or directory
ERROR: can't init develop system, aborting.
````

I can use **./src/darktable --datadir ../data/** but is this correct?
Off-topic: Why is darktable moving database into new XDF directory structure if I have specified a non-default directory/database anyway?

**Comment by codingdave** [comment]
@parafin do you have any idea why these changes are compiling fine on my local machine and failing on travis? Removing the build directory, having a clean git tree does yield working binaries on my side. I can hardly fix this if I cannot reproduce it.

`git status; mkdir build; cd build; cmake -G Ninja -DCMAKE_BUILD_TYPE=RelWithDebInfo ..; ninja; cd ..
On branch fix/CMAKE-remove-unnecessary-code
Your branch is up to date with 'codingdave/fix/CMAKE-remove-unnecessary-code'.

nothing to commit, working tree clean
-- The CXX compiler identification is GNU 9.3.0
`
$ uname -a 
Linux matebook 5.5.13-1-MANJARO #1 SMP PREEMPT Wed Mar 25 17:14:28 UTC 2020 x86_64 GNU/Linux
$ cmake --version
cmake version 3.17.0

CMake suite maintained and supported by Kitware (kitware.com/cmake).

**Comment by codingdave** [comment]
@cytrinox I guess you are confusing this CMAKE fix branch that shall not change the behavior of the build system but make it more compliant to good coding practices for CMAKE with my private branch that allows to run darktable from the build directory. This is not the place to discuss this. Also please don't hijack the PR thread with your question regarding the database.

**Comment by cytrinox** [comment]
@codingdave well I thought this was all related to the out of source tree builds as it seems that this confused darktable about all the paths. Will recheck this without the PR as my assumption was wrong.

**Comment by ptilopteri** [comment]
> How should darktable find the data directory when running from build dir?
> ```
> 
> I can use **./src/darktable --datadir ../data/** but is this correct?
> Off-topic: Why is darktable moving database into new XDF directory structure if I have specified a non-default directory/database anyway?

@cytrinox 
the database location is defined as ~/.config/darktable/library.db, unless you intentionally change it as from the commandline.

**Comment by parafin** [comment]
> @parafin do you have any idea why these changes are compiling fine on my local machine and failing on travis? Removing the build directory, having a clean git tree does yield working binaries on my side. I can hardly fix this if I cannot reproduce it.
> 
> `git status; mkdir build; cd build; cmake -G Ninja -DCMAKE_BUILD_TYPE=RelWithDebInfo ..; ninja; cd ..

I think travis fails in unit tests which are enabled by -DBUILD_TESTING=ON cmake option and you don't have it in your commands. BUILD_TESTING is OFF by default I think.

**Comment by codingdave** [comment]
@parafin unfortunately building with unittests also works fine on my side.
`cmake ../ -G Ninja -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_TESTING=ON`

Any other idea?

**Comment by codingdave** [comment]
I also took special care to be sure that ccache is not being used. I can confirm that ccache is not related to the "false positive" on my side

**Comment by parafin** [comment]
I can't confirm that it doesn't build on my system, because I don't have cmocka installed, but looking at CMakeLists.txt in src/tests I don't see how it can work. Please show full build log with VERBOSE=1 for me to believe it:)

For example src/tests/unittests/iop/test_filmicrgb.c includes src/iop/filmicrgb.c, which in turn includes config.h, which is in build (not source) src/ directory. With your PR it obviously should fail, since you haven't touched src/tests.

Really, you should be able to understand the problem from the travis build logs, and you can reproduce by pushing code to this PR...

**Comment by cytrinox** [comment]
I can reproduce the travis errror (missing config.h) on two machines here when using `cmake ../  -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_TESTING=ON`. Seems like really an issue on your machine/source tree.

I've changed the following:

````
diff --git a/src/tests/CMakeLists.txt b/src/tests/CMakeLists.txt
index c7367bd73..a6ee4e279 100644
--- a/src/tests/CMakeLists.txt
+++ b/src/tests/CMakeLists.txt
@@ -1,3 +1,4 @@
+include_directories("${CMAKE_CURRENT_BINARY_DIR}/../")
 add_executable(darktable-test-variables variables.c)
 target_link_libraries(darktable-test-variables lib_darktable)
````

After that, I compiles fine with testing enabled.

**Comment by codingdave** [comment]
Thanks guys, I am able to reproduce it. Dont know yet whats wrong in my IDE, but I will find out.

**Comment by codingdave** [comment]
@parafin This PR is complete now. Thank you.

**Comment by parafin** [comment]
looks fine

**Comment by TurboGit** [comment]
Ok, so merging now. Thanks for the work @codingdave  and for the review @parafin !

**Pull request merged**